### PR TITLE
fix(core): Return 404 if public key not found

### PR DESCRIPTION
**Pull request by dmihalcik-virtru**
Updates kas_public_key endpoints to return a 404 instead of a 500 if the public key lookup fails. Potential ways a public key is not found:

1. unsupported algorithm
2. unrecognized key identifier (either user supplied TK or default)

This will still 500 on errors in config format or key format

TK should also return 400 on invalid fmt

### CODE REVIEW

**Review by pflynn-virtru — DISMISSED** [review]

**Comment by jakedoublev** [comment]
Will this also fix https://github.com/opentdf/platform/issues/902?

File: .github/workflows/checks.yaml [hunk 1]
```diff
@@ -188,7 +188,11 @@ jobs:
       - run: grpcurl -plaintext localhost:8080 list
       - run: grpcurl -plaintext localhost:8080 grpc.health.v1.Health.Check
       - run: grpcurl -plaintext localhost:8080 kas.AccessService/PublicKey
-      - run: curl --show-error --fail --insecure localhost:8080/kas/v2/kas_public_key
+      - run: curl --show-error --fail-with-body --insecure localhost:8080/kas/v2/kas_public_key
+      - run: curl --show-error --fail-with-body --insecure localhost:8080/kas/v2/kas_public_key?algorithm=ec:secp256r1
+      - run: |-
+          curl_status=$(curl -o /dev/null -s -w "%{http_code}" localhost:8080/kas/v2/kas_public_key?algorithm=invalid)
+          [ $curl_status = 404 ]
```
Comment (strantalis) [inline .github/workflows/checks.yaml:195]: Should we validate the grpc output as well?

Comment (dmihalcik-virtru) [reply .github/workflows/checks.yaml:195]: ok

**Comment by dmihalcik-virtru** [comment]
> Will this also fix #902?

I've added a fix for this as well

**Review by pflynn-virtru — APPROVED** [review]

**Review by strantalis — APPROVED** [review]

**Pull request merged**